### PR TITLE
configure/CMakeLists.txt: Use dpkg-buildflags flags for builds

### LINKED ISSUE #618: Use hardening flags from dpkg-buildpackage for warnings as error jobs
See https://github.com/tango-controls/cppTango/issues/594#issuecomment-549411013.

These flags are on my system

```
$dpkg-buildflags 
CFLAGS=-g -O2 -fdebug-prefix-map=/home/firma/devel/TangoSourceDistribution/test/usr/local/lib=. -fstack-protector-strong -Wformat -Werror=format-security
CPPFLAGS=-Wdate-time -D_FORTIFY_SOURCE=2
CXXFLAGS=-g -O2 -fdebug-prefix-map=/home/firma/devel/TangoSourceDistribution/test/usr/local/lib=. -fstack-protector-strong -Wformat -Werror=format-security
FCFLAGS=-g -O2 -fdebug-prefix-map=/home/firma/devel/TangoSourceDistribution/test/usr/local/lib=. -fstack-protector-strong
FFLAGS=-g -O2 -fdebug-prefix-map=/home/firma/devel/TangoSourceDistribution/test/usr/local/lib=. -fstack-protector-strong
GCJFLAGS=-g -O2 -fdebug-prefix-map=/home/firma/devel/TangoSourceDistribution/test/usr/local/lib=. -fstack-protector-strong
LDFLAGS=-Wl,-z,relro
OBJCFLAGS=-g -O2 -fdebug-prefix-map=/home/firma/devel/TangoSourceDistribution/test/usr/local/lib=. -fstack-protector-strong -Wformat -Werror=format-security
OBJCXXFLAGS=-g -O2 -fdebug-prefix-map=/home/firma/devel/TangoSourceDistribution/test/usr/local/lib=. -fstack-protector-strong -Wformat -Werror=format-security
```

So we could do something like

```
firma@thomas:~/devel/TangoSourceDistribution/test/usr/local/lib (debian-build-fixes)
$dpkg-buildflags --export=sh > test.sh
firma@thomas:~/devel/TangoSourceDistribution/test/usr/local/lib (debian-build-fixes)
$source test.sh 
firma@thomas:~/devel/TangoSourceDistribution/test/usr/local/lib (debian-build-fixes)
```

to use these in one of the warnings as errors travis jobs.

### PR DESCRIPTION
**Pull request by t-b**
We want to be notified on problems via our warnings as errors jobs. And
adding it for the debug build also notifies the developers.

Close #618.

### CODE REVIEW

File: configure/CMakeLists.txt [hunk 1]
```diff
@@ -214,7 +214,7 @@ if(NOT WIN32)
             else()
                 add_compile_options(-O0)
             endif()
-	    add_compile_options(-Wall -Wextra -pedantic)
+            add_compile_options(-Wall -Wextra -pedantic -fstack-protector-strong -Wformat -Werror=format-security)
```
Comment (mliszcz) [inline configure/CMakeLists.txt:217]: Isn't stack-protector something we may consider enabling in release builds, not in debug builds? Per my understanding it is a security feature against attacks based on stack corruption, not a debugging facility. For debugging purposes an address sanitizer job in CI may be another option.

Comment (t-b) [reply configure/CMakeLists.txt]: Good point. I've changed that. I also incorporated the ideas from https://github.com/tango-controls/cppTango/pull/662#discussion_r419463681.

**Review by mliszcz — APPROVED** [review]
LGTM, thanks!

**Comment by t-b** [comment]
Closing as this one is already merged via #662

**Pull request closed without merging**